optuna/trial.py

```diff
@@ -554,35 +554,19 @@ def _is_relative_param(self, name, distribution):
         param_value_in_internal_repr = distribution.to_internal_repr(param_value)
         return distribution._contains(param_value_in_internal_repr)
 
-    def _check_distribution(self, name, low=None, high=None, q=None):
-        # type: (str, Any, Any, Optional[float]) -> None
-
-        dist_dict = {}
-        if low is not None:
-            dist_dict['low'] = low
-        if high is not None:
-            dist_dict['high'] = high
-        if q is not None:
-            dist_dict['q'] = q
-
-        old_distribution_in_trial = self._distributions_in_trial.get(name, None)
-
-        if old_distribution_in_trial is None:
-            self._distributions_in_trial[name] = dist_dict
-        elif old_distribution_in_trial != dist_dict:
-            old_distribution_in_trial_values = 'low = {}, high = {}'.format(
-                old_distribution_in_trial['low'], old_distribution_in_trial['high'])
-            if 'q' in old_distribution_in_trial:
-                old_distribution_in_trial_values += ', q = {}'.format(
-                    old_distribution_in_trial['q'])
+    def _check_distribution(self, name, distribution):
+        # type: (str, BaseDistribution) -> None
+
+        old_distribution = self.distributions.get(name, distribution)
+        if old_distribution != distribution:
             warnings.warn('Inconsistent parameter values for distribution with name "{}"! '
                           'This might be a configuration mistake. '
                           'Optuna allows to call the same distribution with the same '
                           'name more then once in a trial. '
                           'When the parameter values are inconsistent optuna only '
                           'uses the values of the first call and ignores all following. '
                           'Using these values: {}'
-                          .format(name, old_distribution_in_trial_values), RuntimeWarning)
+                          .format(name, str(old_distribution)), RuntimeWarning)
```

Comment:
I'm not sure that it's good to customize `__str__` implemenation for the specific purpose (and I think that there is a room to discuss what is the best string representation of distribution class).
How about using dict representation instead (as in the previous message)? 
```suggestion
                          .format(name, old_distribution._asdict()), RuntimeWarning)
```

Reply:
made this change